### ISSUE TITLE
pins_arduino.h?

### ISSUE THREAD

**Issue by JHDillenb** [dmiddlecamp/particle_arducam#4]
Hello,

I'm trying to compile the code in this repository, but I am getting an error where the ArduCAM.h file has a "#include <pins_arduino.h>" and the pins_arduino.h source file cannot be found. I do not see this pins_arduino.h file within either of the libraries. Is this something that can be downloaded and/or is necessary?

Forgive me if I sound green, I'm still relatively new to the coding world.

**Comment by dmiddlecamp** [comment]
Hi @JHDillenb,

  Some of those include files, like pins_arduino, are included by the development platform you're using.  If you're compiling against a Particle device, they should be excluded automatically, but this repository is rather old.  What device are you compiling against, and what command are you running to do the compile?

Thanks,
David

**Comment by JHDillenb** [comment]
I am compiling against a Photon, device OS 1.3.0-rc.1, using the Particle Workbench to do the compiling, all over VS Code. I have yet to connect the Photon device to my computer or Particle account. Would that cause an issue? 

As of right now with all of my knowledge on the subject, which is very little, I figured that as long as the code compiled without a problem I would be able to flash it to the correct device, regardless of what the device was originally connected to.

Please correct me wherever you see fit.

**Comment by dmiddlecamp** [comment]
Hmm, maybe try changing this line:

https://github.com/dmiddlecamp/particle_arducam/blob/master/firmware/libraries/ArduCAM.h#L75

to something like:

#if defined (PARTICLE)

Sorry that's so out of date!

**Comment by JHDillenb** [comment]
This is rather strange actually.

Not only did I git clone this repository just last week, but even after doing a `git fetch` `git reset --hard origin/master` my ArduCAM.h looks very much different than the one you had linked over to me. Where your `#ifndef ArduCAM_H` is on line 72 mine is on line 95. Much of the code appears similar, but much of it is still different, at least different in ordering from what I've seen. I definitely think I have a different version of the repository. What have I done?? 

The last commit to the repository I have is:
Author: David Middlecamp <david@spark.io>
Date:   Mon May 22 07:57:35 2017 -0700
    adding readme

Does this sound right?

**Comment by dmiddlecamp** [comment]
Hi @JHDillenb ,

  There are a few versions of the library in the repo.  The build script brings in the libraries in "libraries2", sorry about that!

Thanks,
David

**Comment by JHDillenb** [comment]
OHHhh! That's a releif!

Then that would mean that the link you had sent me was the first version, located in "libraries". Does the code use both "libraries" and "libraries2"?

**Comment by dmiddlecamp** [comment]
Sorry about that, it just uses libraries2, you can see it in the build.sh

**Comment by JHDillenb** [comment]
Yes, that's true! However, VS Code is still telling me that `#include <pins_arduino.h>` cannot open its source file. Directly after that, it has in parenthesis `(dependency of "{file location}" `.

Also, just so that you too are aware, I am on the Particle website asking a different question about the same code. The thread is here: https://community.particle.io/t/multiple-definitions-in-particle-arducam/51339

I thought I would spread out my questions to multiple people so not to overwhelm anyone. :)


edit: So I tried deleting the "libraries" folder and compiling it again, and it now compiles successfully!
